controllers/RssController.php

```diff
@@ -0,0 +1,178 @@
+use app\models\Wiki;
+use DateTime;
+use yii\base\Exception;
+use yii\helpers\Url;
+use yii\web\Controller;
+use Zend\Feed\Writer\Feed;
+
+class RssController extends Controller
+{
+    public function actionAll()
+    {
+        $feed = new Feed();
+        // Title of the feed
+        $feed->setTitle('Live News for Yii Framework');
+        // Link to the feed's target, usually a homepage:
+        $feed->setLink('http://www.yiiframework.com/');
+        // Link to the feed itself, and the feed type:
+        $feed->setFeedLink('http://www.yiiframework.com/rss.xml', 'rss');
+
+        // Feed description; only required for RSS:
+        $feed->setDescription('News, fresh extensions and wiki articles about Yii framework.');
+
+        /** @var News[]|Extension[]|Wiki[] $data */
+        $data = array_merge(
+            $this->getNews(),
+            $this->getExtensions(),
+            $this->getWikiPages()
+        );
+
+        asort($data);
+
+        $latest = new DateTime('@0');
+        foreach ($data as $datum) {
+            if ($datum instanceof News) {
+                $entry = $this->createNewsEntry($feed, $datum);
+            } elseif ($datum instanceof Extension) {
+                $entry = $this->createExtensionEntry($feed, $datum);
+            } elseif ($datum instanceof Wiki) {
+                $entry = $this->createWikiEntry($feed, $datum);
+            } else {
+                throw new Exception('Unable to create entry for datum of type ' . get_class($datum));
+            }
+            $feed->addEntry($entry);
+            $latest = $entry->getDateModified() > $latest ? $entry->getDateModified() : $latest;
+        }
+
+        $feed->setDateModified($latest);
+
+        \Yii::$app->response->headers->add('Content-Type', 'application/rss+xml');
+
+        return $feed->export('rss');
+    }
+
+    protected function createNewsEntry(Feed $feed, News $news)
+    {
+        // Create an empty entry:
+        $entry = $feed->createEntry();
+
+        // Set the entry title:
+        $entry->setTitle($news->title);
+
+        // Set the link to the entry:
+        $entry->setLink(Url::toRoute(['news/view', 'id' => $news->id, 'name' => $news->slug], true));
+
+        // Add an author, if you can. Each author entry should be an
+        // array containing minimally a "name" key, and zero or more of
+        // the keys "email" or "uri".
+        $entry->addAuthor(['name' => $news->creator->username]);
+
+        // Set the date created:
+        $entry->setDateCreated(new DateTime($news->created_at));
+
+        // And the date last updated:
+        $modified = new DateTime($news->updated_at);
+        $entry->setDateModified($modified);
+
+        // And finally, some content:
+        $entry->setContent($news->content);
+
+        return $entry;
+    }
+
+    protected function getNews()
+    {
+        return News::find()
+            ->with('creator')
+            ->latest()
+            ->limit(5)
```

Comment:
I'd limit by date, e.g. all news `where('created_at > NOW() - INTERVAL 1 WEEK')` otherwise if 6 new news are added in one day and someone updates the feed the day after, 1 news is missing. Same change should be made for extensions and wiki.